File: src/listOfUsers.java
import java.util.HashMap;

/**
 * Created by Paul Huynh on 11/28/2017.
 * Used to create a local cache of users
 */
public class listOfUsers {

    private HashMap<String,User> Users;                     //Used a hashmap to store user accounts. They're hashed using their email
    private int numOfUsers;                                 //number of users to assign user account numbers


    //initial an empty users list
    listOfUsers(){
        Users = new HashMap<String,User>();
        numOfUsers = 0;
    }



    //function for adding new users
    public boolean addUser(User newUser){
        String userEmail = newUser.getUserEmail();          //get the user's email

        if (Users.containsKey(userEmail))                   //if already exist, send back a false
            return false;

        Users.put(userEmail,newUser);                       //else make the account
        numOfUsers += 1;                                    //increase user count
        return true;                                        //let the main function know it was a success
    }


    //Used this to make sure a user exist and if his password matches
    public boolean checkForUser(String email, String password){
        if(!Users.containsKey(email))
            return false;
        //System.out.println("LOOK: " + Users.get(email).checkPassword(password));
        return Users.get(email).checkPassword(password);
    }


    //Used this function to update a user's information
    public void editUser(String email, User editedUser){
        Users.put(email,editedUser);
    }


    //Used this to return a user by email
    public User getUser(String userEmail){
        User gUser = Users.get(userEmail);
        return gUser;
    }

    //Return back the number of users
    public int getNumOfUsers(){
        return  numOfUsers;
    }

}

File: C#src/ListOfUsers.cs
//Made by Paul Huynh
//ListOfUser class

using System;
using System.Collections;
using System.Collections.Generic;
using System.Text;

namespace BankingLedger
{
    class ListOfUsers
    {
        private Hashtable Users;                        //Hash Table of users, hashed by email
        private int numOfUsers = 0;                     //number of users is used to assign account numbers

        //constructor
        public ListOfUsers()
        {
            numOfUsers = 0;
            Users = new Hashtable();
        }

        //deconstructor
        ~ListOfUsers()
        {
            Users.Clear();
            Users = null;
        }

        //function for adding new users
        public Boolean AddUser(User newUser)
        {
            String userEmail = newUser.GetUserEmail();          //get the user's email

            if (Users.ContainsKey(userEmail))                   //if already exist, send back a false
                return false;

            Users.Add(userEmail, newUser);                       //else make the account
            numOfUsers += 1;                                    //increase user count

            return true;
        }

        //Used this to make sure a user exist and if his password matches
        public Boolean CheckForUser(String email, String password)
        {
            if (!Users.ContainsKey(email))
                return false;
            User u = (User)Users[email];
            return u.CheckPassword(password);
        }

        //Used this function to update a user's information
        public void EditUser(String email, User editedUser)
        {
            Users.Remove(email);
            Users.Add(email, editedUser);
        }

        //Used this to return a user by email
        public User GetUser(String userEmail)
        {
            return (User)Users[userEmail];
        }

        //Return back the number of users
        public int GetNumOfUsers()
        {
            return numOfUsers;
        }

    }
}
